homeassistant/helpers/entity_registry.py

```diff
@@ -168,6 +168,21 @@ def async_update_entity(self, entity_id, *, name=_UNDEF,
             new_entity_id=new_entity_id
         )
 
+    @callback
+    def async_migrate_entity_unique_id(self, from_unique_id, to_unique_id):
+        """Migrate unique_id of a device."""
+        conflict = next((entity for entity in self.entities.values()
+                         if entity.unique_id == to_unique_id), None)
+        if conflict:
+            raise ValueError("Unique id '{}' is already in use by '{}'".format(
+                to_unique_id, conflict.entity_id))
+        old = next(entity for entity in self.entities.values()
```

Comment:
Uniqueness is combination of platform, domain and unique id

Reply:
Done.  I also moved this to within the existing `update_entity` method (to be parallel with the device registry) so I don't have to duplicate the subscription updates.  It's fine having to resolve the `entity_id` first in order to call this.